src/ui/render.ts

```diff
@@ -159,15 +207,21 @@ export default class UI {
     cellDom.id = `${colIndex}_${rowIndex}`;
     const inputDom = document.createElement("input");
     inputDom.classList.add("lightsheet_table_cell_input");
-    inputDom.value = "";
     inputDom.readOnly = this.isReadOnly;
     cellDom.appendChild(inputDom);
 
-    if (value) {
+    if (rawValue) {
       cellDom.id = `${columnKey}_${rowDom.id}`;
-      inputDom.value = value;
     }
 
+    if (this.lightSheet.isReady)
+      this.onUICellValueChange(rawValue, colIndex, rowIndex);
```

Comment:
This was the max callstack issue.
I had to put the if here, otherwise the browser goes to an event callback loop.
@Fluglow you can try removing this and please do let us know if this is ok.

Fyi the loop scenario I was able to generate was if you put and empty string as the first cell value of a row.